tests/upgrade/0.8.2-0.9.0/jiva/suite_test.go

```diff
@@ -0,0 +1,234 @@
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jiva
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	unstruct "github.com/openebs/maya/pkg/unstruct/v1alpha2"
+	http "github.com/openebs/maya/pkg/util/http/v1alpha1"
+	"github.com/openebs/maya/tests"
+	"github.com/openebs/maya/tests/artifacts"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	// auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var (
+	kubeConfigPath        string
+	replicaCount          int
+	nsName                = "default"
+	scName                = "jiva-upgrade-sc"
+	openebsProvisioner    = "openebs.io/provisioner-iscsi"
+	replicaLabel          = "openebs.io/replica=jiva-replica"
+	ctrlLabel             = "openebs.io/controller=jiva-controller"
+	openebsCASConfigValue = "- name: ReplicaCount\n  Value: "
+	accessModes           = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	capacity              = "5G"
+	pvcObj                *corev1.PersistentVolumeClaim
+	pvcName               = "jiva-volume-claim"
+	scObj                 *storagev1.StorageClass
+	openebsYAML,
+	rbacYAML,
+	crYAML,
+	runtaskYAML,
+	jobYAML string
+)
+
+func TestSource(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test jiva volume upgrade")
+}
+
+func init() {
+	flag.StringVar(&kubeConfigPath, "kubeconfig", "", "path to kubeconfig to invoke kubernetes API calls")
+	flag.IntVar(&replicaCount, "replicas", 1, "number of replicas to be created")
+}
+
+var ops *tests.Operations
+
+var _ = BeforeSuite(func() {
+
+	openebsYAML = getYAMLFromURL("https://openebs.github.io/charts/openebs-operator-0.8.2.yaml")
+	rbacYAML = getYAMLFromURL("https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/rbac.yaml")
+	crYAML = getYAMLFromURL("https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/jiva/cr.yaml")
+	runtaskYAML = getYAMLFromURL("https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/jiva/jiva_upgrade_runtask.yaml")
+	jobYAML = getYAMLFromURL("https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/jiva/volume-upgrade-job.yaml")
```

Comment:
I guess, we can create the job via builder pattern.
We create a jiva volume on 0.8.2
Then build a job using above volume name, namespace, etc.
Then upgrade jiva volume via above job.

Reply:
We don't need to build the job as it remains same all the time. We have to build the ConfigMap as the data field of ConfigMap contains the name and namespace of pv. That too building the object is not required. We can just set the data field of ConfigMap before applying it.

Reply:
Besides that any changes made in the yamls will reflect automatically. We should not build objects to hard code things.